### PR TITLE
Allow zero hydro bodies resolves issue #655

### PR DESCRIPTION
**Pull request by akeeste**
This PR updates the body logic in WEC-Sim preprocessing to allow for zero hydro bodies. This condition is required if a user wants to run a case that only contains traditional Morison Elements. 

If no hydro body is used (and therefore no .h5 file is read), users must define the ``waves.freqRange`` and ``waves.waterDepth`` variables. As with the current dev branch, users must define the cg and displace volume of drag/nonhydro bodies in the input file. Checks for these conditions are updated. 

This resolves issue #655

### CODE REVIEW

**Comment by kmruehl** [comment]
@akeeste do you have an example to test this with? It could be added to the https://github.com/WEC-Sim/WEC-Sim_Applications repository.

**Comment by akeeste** [comment]
@kmruehl My seedling would be relevant, but that is several months away from becoming an application case. I could make a simplified example. Also I admit I still need to go through the documentation to update any comments requiring 1+ hydro body. I'll make this a draft PR until that is done.

**Comment by akeeste** [comment]
This PR is now ready for review. I have updated the documentation to reflect that hydro bodies are no longer required for WEC-Sim to run (though at least one type of body is, regardless of type). [Application repo PR #11](https://github.com/WEC-Sim/WEC-Sim_Applications/pull/11) contains an application case comparing a traditional Morison Element (no hydro bodies) to a WEC-Sim hydrodynamic body.

File: source/objects/bodyClass.m [hunk 1]
```diff
@@ -186,12 +186,20 @@ function loadHydroData(obj, hydroData)
...
+        end
+        
+        function dragForcePre(obj,rho,nlHydro)
             % DragBody Pre-processing calculations
             % Similar to hydroForcePre, but only loads in the necessary
             % values to calculate linear damping and viscous drag. Note
             % that body DOF is inherited from the length of the drag
             % coefficients.
+            obj.setMassMatrix(rho,nlHydro);
             if  any(any(obj.viscDrag.Drag)) == 1  %check if obj.viscDrag.Drag is defined
```
Comment (kmruehl) [inline source/objects/bodyClass.m:203]: Verify compatibility with https://github.com/WEC-Sim/WEC-Sim/pull/692

File: source/objects/bodyClass.m [hunk 2]
```diff
@@ -720,16 +728,18 @@ function setMassMatrix(obj, rho, nlHydro)
             % This method sets mass for the special cases of body at equilibrium or fixed and is used by hydroForcePre.
             if strcmp(obj.mass, 'equilibrium')
                 obj.massCalcMethod = obj.mass;
-                if nlHydro == 0
+                if obj.nhBody == 0 && nlHydro == 0
                     obj.mass = obj.hydroData.properties.disp_vol * rho;
-                else
+                elseif obj.nhBody == 0 && nlHydro ~= 0
                     cg_tmp = obj.hydroData.properties.cg;
```
Comment (kmruehl) [inline source/objects/bodyClass.m:734]: This needs to be changed based on https://github.com/WEC-Sim/WEC-Sim/pull/692

**Review by kmruehl — APPROVED** [review]
@akeeste this should be ready for a merge, but you may want to review it one more time before I merge it.

**Comment by akeeste** [comment]
@kmruehl Thanks for updating this! I just reviewed one more time and things look good

**Pull request merged**